Panic if receive a UpdateL1InfoTreeV2 without previous UpdateL1InfoTree

#### System information
Version: `v0.1.2`

````
2024-10-03T17:20:32.965Z	INFO	l1_sync/l1_syncer_sequential.go:150	Syncing FromBlock: 1630, ToBlock: 1639, MaximumBlock: 1639 Progress: Percent: 99.5	{"pid": 9}
2024-10-03T17:20:32.965Z	INFO	l1_sync/l1_syncer_sequential.go:211	Getting rollup info range: FromBlock: 1631, ToBlock: 1639 overlappedFirstBlock:false finalizedBlock:1639	{"pid": 9}
2024-10-03T17:20:32.967Z	INFO	etrog/processor_l1_info_tree_update.go:48	L1InfoTree(etrog) stored. BlockNumber: 1631,GER:0x2a3047559f9342b54967012179a6b33eb87aee146f7f8539f3d03af1f435a416 L1InfoTreeIndex: 97 L1InfoRoot:0x6a9a19181079d0784727d21c749f4098f5052ef9c74e9fd9b282c7492d6b8db7 event_data:BlockNumber: 1631 MainnetExitRoot: 0xf7fdd23bf683e6c9988dfca0a161627e39e8ddeb63da0be347400528351860c1 RollupExitRoot: 0x27ae5ba08d7291c96c8cbddcc148bf48a6d68c7974b94356f53754ef6171d757 GlobalExitRoot: 0x2a3047559f9342b54967012179a6b33eb87aee146f7f8539f3d03af1f435a416 Timestamp: 2024-10-03 17:20:24 +0000 UTC PreviousBlockHash: 0x5e6c085eaa6d56a6501ff3e6efa90cf07ff80613d1d2ce23e1c7640692e4aa65	{"pid": 9}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x11f8661]

goroutine 265 [running]:
github.com/0xPolygonHermez/zkevm-synchronizer-l1/synchronizer/actions/banana.(*ProcessorUpdateL1InfoTreeV2).ProcessUpdateL1InfoTreeV2(0xc010c14cf0, {0x1e79920, 0xc010b3cc30}, 0xc00004d758?, {{0x8d, 0x7f, 0x76, 0x1c, 0x91, 0xba, ...}, ...}, ...)
	/go/pkg/mod/github.com/0x!polygon!hermez/zkevm-synchronizer-l1@v1.0.1/synchronizer/actions/banana/processor_update_l1_info_tree_v2.go:50 +0x141
github.com/0xPolygonHermez/zkevm-synchronizer-l1/synchronizer/actions/banana.(*ProcessorUpdateL1InfoTreeV2).Process(0x19a4b9e?, {0x1e79920?, 0xc010b3cc30?}, 0x1e7a878?, {{0x19a4b9e?, 0x1?}, 0x100ffffffff?}, 0x7f1de3583488?, {0x1e7a878, 0xc000d56600})
	/go/pkg/mod/github.com/0x!polygon!hermez/zkevm-synchronizer-l1@v1.0.1/synchronizer/actions/banana/processor_update_l1_info_tree_v2.go:38 +0xf8
github.com/0xPolygonHermez/zkevm-synchronizer-l1/synchronizer/actions/processor_manager.(*L1EventProcessors).Process(0x6?, {0x1e79920, 0xc010b3cc30}, 0xc, {{0x19a4b9e?, 0x1e77dc0?}, 0x0?}, 0xc000dc6388, {0x1e7a878, 0xc000d56600})
	/go/pkg/mod/github.com/0x!polygon!hermez/zkevm-synchronizer-l1@v1.0.1/synchronizer/actions/processor_manager/processor_manager.go:69 +0x1e8
github.com/0xPolygonHermez/zkevm-synchronizer-l1/synchronizer/internal.(*BlockRangeProcess).processElement(0xc00081de80, {0x1e79920, 0xc010b3cc30}, {{0x19a4b9e?, 0x38?}, 0x1942140?}, {0xc000dc6388, 0xc00004da20?, 0x11503fc?}, 0x0, ...)
	/go/pkg/mod/github.com/0x!polygon!hermez/zkevm-synchronizer-l1@v1.0.1/synchronizer/internal/synchronizer_block_range_process.go:145 +0x3c4
github.com/0xPolygonHermez/zkevm-synchronizer-l1/synchronizer/internal.(*BlockRangeProcess).processBlock(0xc00081de80, {0x1e79920, 0xc010b3cc30}, {0xc000dc6388, 0x2, 0x2}, 0x0, {0x1e7a878, 0xc000d56600}, 0xc000824a50, ...)
	/go/pkg/mod/github.com/0x!polygon!hermez/zkevm-synchronizer-l1@v1.0.1/synchronizer/internal/synchronizer_block_range_process.go:122 +0x28c
github.com/0xPolygonHermez/zkevm-synchronizer-l1/synchronizer/internal.(*BlockRangeProcess).internalProcessBlockRange(0xc00081de80, {0x1e79920, 0xc010b3cc30}, {0xc000dc6388, 0x2, 0x2}, 0xc000824a50, 0x667, 0x1, {0x7f1de35cf270, ...})
	/go/pkg/mod/github.com/0x!polygon!hermez/zkevm-synchronizer-l1@v1.0.1/synchronizer/internal/synchronizer_block_range_process.go:95 +0x165
github.com/0xPolygonHermez/zkevm-synchronizer-l1/synchronizer/internal.(*BlockRangeProcess).ProcessBlockRange(0xc010b66480?, {0x1e79920?, 0xc010b3cc30?}, {0xc000dc6388?, 0x2?, 0x0?}, 0x0?, 0x1?)
	/go/pkg/mod/github.com/0x!polygon!hermez/zkevm-synchronizer-l1@v1.0.1/synchronizer/internal/synchronizer_block_range_process.go:52 +0x3c
github.com/0xPolygonHermez/zkevm-synchronizer-l1/synchronizer/l1_sync.(*L1SequentialSync).iteration(0xc010b66480, {0x1e79920, 0xc010b3cc30}, {0x2?, 0x2?, 0x2?}, 0x667, 0xc0083ee080)
	/go/pkg/mod/github.com/0x!polygon!hermez/zkevm-synchronizer-l1@v1.0.1/synchronizer/l1_sync/l1_syncer_sequential.go:229 +0x38d
github.com/0xPolygonHermez/zkevm-synchronizer-l1/synchronizer/l1_sync.(*L1SequentialSync).SyncBlocks(0xc010b66480, {0x1e79920, 0xc010b3cc30}, 0xc0083ee080)
	/go/pkg/mod/github.com/0x!polygon!hermez/zkevm-synchronizer-l1@v1.0.1/synchronizer/l1_sync/l1_syncer_sequential.go:152 +0x411
github.com/0xPolygonHermez/zkevm-synchronizer-l1/synchronizer/internal.(*SynchronizerImpl).Sync(0xc010b483c0, 0x0)
	/go/pkg/mod/github.com/0x!polygon!hermez/zkevm-synchronizer-l1@v1.0.1/synchronizer/internal/synchronizer_impl.go:259 +0x3b5
github.com/0xPolygonHermez/zkevm-synchronizer-l1/synchronizer.(*SynchronizerAdapter).Sync(0x0?, 0x0?)
	/go/pkg/mod/github.com/0x!polygon!hermez/zkevm-synchronizer-l1@v1.0.1/synchronizer/synchronizer_adapter.go:44 +0x1a
github.com/0xPolygon/cdk/aggregator.(*Aggregator).Start.func1()
	/go/src/github.com/0xPolygon/cdk/aggregator/aggregator.go:669 +0x31
created by github.com/0xPolygon/cdk/aggregator.(*Aggregator).Start in goroutine 213
	/go/src/github.com/0xPolygon/cdk/aggregator/aggregator.go:668 +0x150
````

When submitting logs: please submit them as text and not screenshots.

--- a/synchronizer/actions/banana/processor_update_l1_info_tree_v2.go
+++ b/synchronizer/actions/banana/processor_update_l1_info_tree_v2.go
@@ -47,10 +47,15 @@ func (p *ProcessorUpdateL1InfoTreeV2) ProcessUpdateL1InfoTreeV2(ctx context.Cont
 		log.Errorf("error getting the state leaf. Error: %v", err)
 		return err
 	}
-	err = compareL1InfoTreeLeaf(*stateLeaf, data)
-	if err != nil {
-		log.Errorf("error comparing the state leaf. Error: %v", err)
-		return err
+	if stateLeaf != nil {
+		err = compareL1InfoTreeLeaf(*stateLeaf, data)
+		if err != nil {
+			log.Errorf("error comparing the state leaf. Error: %v", err)
+			return err
+		}
+		log.Infof("L1InfoTreeLeafV2 sanity check OK: %s", data.String())
+	} else {
+		log.Warnf("this l1nfotree is not stored on local DB. So can't check it: data:%s ", data.String())
 	}
 	return nil
 }

--- a/version.go
+++ b/version.go
@@ -7,7 +7,7 @@ import (
 )
 
 var (
-	Version = "v1.0.2"
+	Version = "v1.0.3"
 )
 
 // PrintVersion prints version info into the provided io.Writer.

